src/behavioural-events/events-tracker.ts

```diff
@@ -96,6 +104,12 @@ export default class EventsTracker implements IEventsTracker {
 
     const eventsToFlush = [...this.eventsQueue];
 
+    if (this.isSilent) {
+      Logger.debugLog("Skipping events tracking, the EventsTracker is silent");
+      this.eventsQueue.splice(0, eventsToFlush.length);
```

Comment:
Hmm would it make more sense to not add the events to the queue in the first place instead? If we want to optimize it, we could even just not create the `EventTracker` in the first place, and make it optional inside `Purchases`.

This is considered this setting can't be changed after configuration.

Reply:
I was taking another look, and my idea to not create the EventTracker in the first place is trickier, since this class also generates the `traceId` which gets sent to the backend when starting a checkout session. I think it might be better to leave that, but lmk if we would prefer to stop sending that property as well